react/__tests__/SliderTrack.test.tsx

```diff
@@ -0,0 +1,272 @@
+import React from 'react'
+import { render, fireEvent } from '@vtex/test-tools/react'
+
+import SliderTrack from '../components/SliderTrack'
+import { useSliderState } from '../components/SliderContext'
+import {
+  mockInitialInfiniteSliderState,
+  mockInitialNonInfiniteSliderState,
+} from '../__fixtures__/SliderStateContext'
+
+const mockDispatch = jest.fn()
+
+const mockedUseSliderState = useSliderState as jest.Mock
+
+jest.mock('../components/SliderContext', () => ({
+  useSliderState: jest.fn(),
+  useSliderDispatch: () => mockDispatch,
+}))
+
+describe('Basic rendering', () => {
+  mockedUseSliderState.mockImplementation(() => mockInitialInfiniteSliderState)
+
+  it('should render with correct width based on slides to render', () => {
+    // slidesPerPage < totalItems
+    const { getByTestId, rerender } = render(
+      <SliderTrack usePagination totalItems={10} infinite />
+    )
+
+    let renderedSliderTrack = getByTestId('slider-track')
+
+    expect(renderedSliderTrack.style.width).toEqual(`${(20 * 100) / 5}%`)
```

Comment:
Maybe it would be nice to add a comment here explaining why it should be `20 * 100 / 5`. Something small like "If we have x slides and the page size is y, the `sliderTrack` should have z% of width of the slider window

Reply:
Sure!!